src/platform_impl/macos/util/async.rs

```diff
@@ -171,7 +171,9 @@ pub unsafe fn set_maximized_async(
       if shared_state_lock.fullscreen.is_some() {
         // Handle it in window_did_exit_fullscreen
         return;
-      } else if curr_mask.contains(NSWindowStyleMask::NSResizableWindowMask) {
+      } else if curr_mask.contains(NSWindowStyleMask::NSResizableWindowMask)
+        && curr_mask.contains(NSWindowStyleMask::NSTitledWindowMask)
```

Comment:
If window has not zoom button, `ns_window.zoom_(nill)` can not zoom window.
See https://developer.apple.com/documentation/appkit/nswindow/1419513-zoom